lib/mongo/uri.rb

```diff
@@ -12,7 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-require 'uri'
+require 'addressable'
```

Comment:
hm, I really don't want a dependency on another gem for this. Are there any other solutions?